include/imterm/terminal.hpp

```diff
@@ -80,6 +80,10 @@ namespace ImTerm {
 		using buffer_type = std::array<char, 1024>;
 		using small_buffer_type = std::array<char, 128>;
 
+		std::atomic_flag m_flag;
```

Comment:
Shouldn’t `m_flag` be cleared in the constructor though ? According to [cppreference](https://en.cppreference.com/w/cpp/atomic/atomic_flag/atomic_flag) the constructor initializes it to an unspecified state prior C++20.

Also, please move its declaration at the end of the class (with the other variable declarations)

Reply:
@Organic-Code Sorry, I didn't notice the difference as I was using c++ 20 in development. Let me update it to your requirement.